atomic_reactor/plugins/pre_add_image_content_manifest.py

```diff
@@ -83,16 +83,24 @@ class AddImageContentManifestPlugin(PreBuildPlugin):
         'image_contents': [],
     }
 
-    def __init__(self, tasker, workflow, remote_source_icm_url=None, destdir=IMAGE_BUILD_INFO_DIR):
+    def __init__(self, tasker, workflow, remote_sources=None, destdir=IMAGE_BUILD_INFO_DIR):
         """
         :param tasker: ContainerTasker instance
         :param workflow: DockerBuildWorkflow instance
-        :param icm_url: str, URL of the ICM from the Cachito request.
+        :param remote_sources: list of dicts, each dict contains info about particular
+        remote source with the following keys:
+            build_args: dict, extra args for `builder.build_args`, if any
+            configs: list of str, configuration files to be injected into
+            the exploded remote sources dir
+            icm_url: str, the Cachito ICM URL; used to request the
+            Image Content Manifest
+            url: str, URL from which to download a source archive
+            name: str, name of remote source
```

Comment:
Suggest to create a "remote source" object.

Reply:
That is not a bad idea, but this was the decision taken in the design

Reply:
Improvements over the design are always welcome :wink: 

I think the param has to be a dict though. Correct me if I'm wrong, but I think all plugin params must be JSON-compatible objects because that's how osbs-client passes them to atomic-reactor. You could convert the param into some kind of "remote source" object inside the plugin though, I'll leave that up to you.

Reply:
Yup, it has to be JSONizable (if that is a word, JSON serializable)

Reply:
I am going to keep it as dict for now, unless someone is strongly against.